src/Services/ItemService.cs

```diff
@@ -1,19 +1,44 @@
 ﻿using DomainModels;
+using Repositories;
 using Services.DTO;
 using System.Collections.Generic;
 
 namespace Services
 {
-    public class ItemService
+    public class ItemService : IItemService
     {
-        public int Get(int id)
+        private readonly IItemRepository _itemRepository;
+
+        public ItemService(IItemRepository itemRepository)
+        {
+            _itemRepository = itemRepository ?? throw new System.ArgumentNullException(nameof(itemRepository));
+        }
+
+        public ItemDTO Get(int id)
+        {
+            return new ItemDTO { Text = "This is a sample text" };
+        }
+
+        public IEnumerable<ItemDTO> GetAll()
+        {
+            return new List<ItemDTO>() { new ItemDTO { Text = "Sanple Text yessir" } };
+        }
+
+        public IEnumerable<ItemDTO> GetAllByFilter(ItemByFilterDTO filters)
+        {
+            return new List<ItemDTO>() { new ItemDTO { Text = "Sanple Text yessir" } };
+        }
+
+        public void AddItem(ItemDTO itemDto)
         {
-            return id;
+
         }
+        public void Update(ItemDTO itemDto)
+        {
 
-        public IEnumerable<string> GetAll(int id)
+        }
+        public void Delete(int id)
         {
-            return new List<string>() { "item 1" };
```

Comment:
Please implement the method using the injected IItemRepository.

Reply:
implemented